### PR TITLE
91354686 oe person

### PR DESCRIPTION
**Pull request by jorgelopezlago**
Doesn't include tests.
Person adds relationship fields for performers to Event content type.

### CODE REVIEW

**Comment by samueljmurray** [comment]
@jorgelopezlago I pulled this branch down to do the CleanUp function in the test, but found some issues. After running a clean installation against the branch and running the tests, I found:

- the Event–Person relation isn't installed
- the fields on the Person content type aren't in field groups
- the 'Performers' field on the Event content type doesn't sit in a field group
- the tests fail. The first error is:

```
1) The Person features of the CMS can create, edit and delete a person
   Message:
     Expected 'Error message
Title field is required.' to contain 'Person Jon Snow has been created.'.
   Stacktrace:
     Error: Failed expectation
    at [object Object].<anonymous> (/Users/smurray/Workspace/southbankcentre.org-CMS/test/functional/person/person-spec.js:127:32)
```

**Comment by jorgelopezlago** [comment]
@samueljmurray I tried to reproduce your problems and I could not do it.
On a clean install:
1) Can not reproduce it (installed correctly on my tests)
2) Can not reproduce it (fields are in the correct vertical tab)
3) Can not reproduce it (fields in the correct vertical tab)
4) Fixed as part of OE Event 7.x-1.2
5) Can not reproduce. I had mixed results with occasional failures on the last message after Person creation: 
> 1) The Person features of the CMS sets relationship to Event
   Message:
     NoSuchElementError: No element found using locator: By.cssSelector(".messages")

**Comment by samueljmurray** [comment]
@jorgelopezlago sorry, it was a problem with my permissions settings which meant some files didn't update when I pulled the latest code. Fixed now

**Comment by tassosk-cogapp** [comment]
@jorgelopezlago this is back to you. Sam's clean up function need some love. Some comments on the pull request:
- can you make sure that all strings in `$vocab` and `$terms` in person.install are wrapper in `t()` functions.

**Comment by jorgelopezlago** [comment]
@tassosk-cogapp I am not wrapping the strings in `$vocab` and `$terms` in `t()` because they are not part of the interface of Drupal, just auxiliary data structures to create *testing content strings* that do not need to be registered in the translation system as translatable.

In any case in an install file we should use `st()` instead of `t()` as the translation system may not be available and we have to read from PO files (if available): https://api.drupal.org/api/drupal/includes%21install.inc/function/st/7

**Comment by tassosk-cogapp** [comment]
I agree with the st(). There are a few instances that t() is used so I guess those need to be migrated to st() too . https://www.drupal.org/node/299085.

File: drupal/profiles/openevent/modules/features/oe_event/oe_event.info [hunk 1]
```diff
@@ -2,7 +2,7 @@ name = OE Event
 description = Provides configuration allowing representation of the events.
 core = 7.x
 package = Open Events
-version = 7.x-1.0
+version = 7.x-1.1
```
Comment (alexbridge) [inline drupal/profiles/openevent/modules/features/oe_event/oe_event.info]: Are we happy that when this gets a more formal public release, the module versions won't be beginning at 1.0?

File: test/functional/event/event-spec.js [hunk 2]
```diff
@@ -164,7 +164,7 @@ describe('The Event features of the CMS', function() {
     // fill out content on 'Main' tab
     dvr.findElement(by.xpath("//ul[@class='vertical-tabs-list']/li/a[strong='Main']")).click();
     dvr.findElement(by.id('edit-title')).sendKeys('Protractor event page');
-    dvr.findElement(by.id('edit-field-teaser-und-0-value')).sendKeys('Here is some content in the teaser field <em>that contains emphasis</em> but <script>doesNotContainJavascript();</script>');
+    //dvr.findElement(by.id('edit-field-teaser-und-0-value')).sendKeys('Here is some content in the teaser field <em>that contains emphasis</em> but <script>doesNotContainJavascript();</script>');
```
Comment (alexbridge) [inline test/functional/event/event-spec.js]: Can this be removed completely? Or should it be uncommented?

File: test/functional/event/event-spec.js [hunk 3]
```diff
@@ -311,10 +311,10 @@ describe('The Event features of the CMS', function() {
       .expectStatus(200)
       .expectHeaderContains('content-type', 'application/json')
       .expectJSON('list.0', {
-        "field_teaser": {
-          "value": "<p>Here is some content in the teaser field <em>that contains emphasis</em> but doesNotContainJavascript();</p>\n",
-          "format": "filtered_html"
-        },
+        //"field_teaser": {
```
Comment (alexbridge) [inline test/functional/event/event-spec.js]: Can this be removed completely? Or should it be uncommented?

File: test/functional/person/person-spec.js [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+  var pageTitle = element(by.css('.page-title'));
+  var save = element(by.id('edit-submit'));
+  var del = element(by.id('edit-delete'));
+  var messages = element(by.css('.messages')); // How do we distinguish multiple messages?
+
+  // Tab Main
+  var tabMain = element(by.xpath("//ul[@class='vertical-tabs-list']/li/a[strong='Main']"));
+
+  // Main fields
+  // @TODO add a rand to get one of these
```
Comment (alexbridge) [inline test/functional/person/person-spec.js]: Is this todo for a later point, needed now, or no longer needed?

File: test/functional/person/person-spec.js [hunk 5]
```diff
@@ -0,0 +1,461 @@
...
+
+  it('can create, edit and delete a person', function() {
+    browser.get(browser.params.url + '/node/add/person');
+    expect(pageTitle.getText()).toContain('Create Person');
+
+    // Fill in person details
+    personGivenName.sendKeys('Jon');
+    personFamilyName.sendKeys('Snow');
+    personNameSuffix.sendKeys('Commander');
+    personAlias.sendKeys('Ned Stark\'s bastard son');
```
Comment (alexbridge) [inline test/functional/person/person-spec.js]: !

File: test/functional/person/person-spec.js [hunk 6]
```diff
@@ -0,0 +1,461 @@
...
+            }
+          ]
+        })
+      .after(cleanUp)
+      .toss();
+
+    // We run cleanUp after the last frisby test because they are asynchronous
+    // and could run after the cleanUp otherwise (this is something to improve on)
+    function cleanUp () {
+      // These tests are destructive and can only be performed on clean/empty sites [!]
```
Comment (alexbridge) [inline test/functional/person/person-spec.js]: Can commented-out lines in this section be removed?

File: drupal/profiles/openevent/modules/features/oe_event/oe_event.info (hunk 1 above)
Comment (jorgelopezlago) [inline drupal/profiles/openevent/modules/features/oe_event/oe_event.info]: Too late for this now. I do not think there is much problem with a starting public release beyond 1.0. Not desirable, but..
 We could also move the version back to 0.x instead. That doesn't look right either.

File: test/functional/event/event-spec.js (hunk 2 above)
Comment (jorgelopezlago) [inline test/functional/event/event-spec.js]: It can be removed. It is duplicated.

File: test/functional/event/event-spec.js (hunk 3 above)
Comment (jorgelopezlago) [inline test/functional/event/event-spec.js]: I guess you want the code to be pretty tidy, eh?
We are not testing for teaser now, and it is a field that has been removed form OE. To be removed.

File: test/functional/person/person-spec.js (hunk 5 above)
Comment (jorgelopezlago) [inline test/functional/person/person-spec.js]: Greek testing, raw!
Being the Open distribution we should remove the swearing.

File: test/functional/person/person-spec.js (hunk 6 above)
Comment (jorgelopezlago) [inline test/functional/person/person-spec.js]: Why?

Comment (alexbridge) [inline test/functional/person/person-spec.js]: The excerpt above doesn't show what I'm referring to: there are a number of commented-out code lines that it's not clear whether they're redundant or need to be re-added.

File: test/functional/person/person-spec.js (hunk 4 above)
Comment (jorgelopezlago) [inline test/functional/person/person-spec.js]: I think this one doesn't harm. It looks like a nice to have... for the future.

**Pull request merged**